c-basics/src/isOlder.cpp

```diff
@@ -41,89 +41,95 @@ bool validatedate(char *dob1, char *dob2, int *d1, int *d2)
+	*d2 = ((*dob2 - 48) * 10) + (*(dob2 + 1) - 48);
 
-		*m1 = (((*(dob1 + 3) - 48) * 10) + (*(dob1 + 4) - 48));
-		*m2 = (((*(dob2 + 3) - 48) * 10) + (*(dob2 + 4) - 48));
-		*y1 = (((((((*(dob1 + 6) - 48) * 10) + (*(dob1 + 7) - 48)) * 10) + (*(dob1 + 8) - 48)) * 10) + *(dob1 + 9) - 48);
+	*m1 = (((*(dob1 + 3) - 48) * 10) + (*(dob1 + 4) - 48));
+	*m2 = (((*(dob2 + 3) - 48) * 10) + (*(dob2 + 4) - 48));
 
-		*y2 = (((((((*(dob2 + 6) - 48) * 10) + (*(dob2 + 7) - 48)) * 10) + (*(dob2 + 8) - 48)) * 10) + *(dob2 + 9) - 48);
-		if ((*d1 > 31 || *d2 > 31) || (*m1 > 12 || *m2 > 12))
-			return false;
-		if (*m1 == 2)
+	*y1 = (((((((*(dob1 + 6) - 48) * 10) + (*(dob1 + 7) - 48)) * 10) + (*(dob1 + 8) - 48)) * 10) + *(dob1 + 9) - 48);
+	*y2 = (((((((*(dob2 + 6) - 48) * 10) + (*(dob2 + 7) - 48)) * 10) + (*(dob2 + 8) - 48)) * 10) + *(dob2 + 9) - 48);
+}
+
+bool checkForFebruaryDates(int y1, int y2, int d1, int d2, int m1, int m2)
+{
+	if ((d1 > 31 || d2 > 31) || (m1 > 12 || m2 > 12))
+		return false;
+	if (m1 == 2)
+	{
+		if (y1 % 400 == 0 || (y1 % 100 != 0 && y1 % 4 == 0))
 		{
-			if (*y1 % 400 == 0 || (*y1 % 100 != 0 && *y1 % 4 == 0))
-			{
-				//printf("i");
-				if (*d1 > 29)
-					return false;
-			}
-			else if ((*y1 % 4 != 0) && *d1>28)
+			if (d1 > 29)
 				return false;
 		}
-		if (*m2 == 2)
+		else if ((y1 % 4 != 0) && d1>28)
+			return false;
+	}
+	if (m2 == 2)
+	{
+		if (y2 % 400 == 0 || (y2 % 100 != 0 && y2 % 4 == 0))
 		{
-			//printf("i");
-			if (*y2 % 400 == 0 || (*y2 % 100 != 0 && *y2 % 4 == 0))
-			{
-				//printf("l");
-				if (*d2 > 29)
-					return false;
-			}
-			else if ((*y2 % 4 != 0) && *d2>28)
+			if (d2 > 29)
 				return false;
 		}
-		return true;
+		else if ((y2 % 4 != 0) && d2>28)
+			return false;
 	}
-	else return false;
 }
 
 int isOlder(char *dob1, char *dob2)
 {
 	int year1, year2, date1, date2, month1, month2;
-	if ((validatedate(dob1, dob2, &date1, &date2) && validatemonth(dob1, dob2, &month1, &month2)) && validateyear(dob1, dob2, &year1, &year2, &date1, &date2, &month1, &month2))
+	if (validateDateFormat(dob1, dob2) && validateMonthFormat(dob1, dob2) && validateYearFormat(dob1, dob2))
 	{
-		if (year1 > year2)
+		getDateMonthYear(dob1, dob2, &year1, &year2, &date1, &date2, &month1, &month2);
+		if (checkForFebruaryDates(year1, year2, date1, date2, month1, month2))
 		{
-			return 2;
-		}
-		else if (year2 > year1)
-		{
-			return 1;
-		}
-		else
-		{
-			if (month1 > month2)
+			if (year1 > year2)
 			{
 				return 2;
 			}
-			else if (month2 > month1)
+			else if (year2 > year1)
 			{
 				return 1;
 			}
 			else
 			{
-				if (date1 > date2)
+				if (month1 > month2)
 				{
 					return 2;
 				}
-				else if (date2 > date1)
+				else if (month2 > month1)
 				{
 					return 1;
 				}
-				else return 0;
+				else
+				{
+					if (date1 > date2)
+					{
+						return 2;
+					}
+					else if (date2 > date1)
+					{
+						return 1;
+					}
+					else return 0;
+				}
 			}
 		}
-
```

Comment:
Looks good for now. From next make sure that the code looks as modular as possible.